libbeat/publisher/preprocess.go

```diff
@@ -158,14 +158,13 @@ func updateEventAddresses(publisher *PublisherType, event common.MapStr) bool {
 			//outgoing transaction
 			event["direction"] = "out"
 		}
-	}
 
-	if publisher.IgnoreOutgoing && dstServer != "" &&
-		dstServer != publisher.name {
-		// duplicated transaction -> ignore it
-		debug("Ignore duplicated transaction on %s: %s -> %s",
-			publisher.name, srcServer, dstServer)
-		return false
+		if publisher.IgnoreOutgoing && event["direction"] == "out" {
```

Comment:
move the check for IgnoreOutgoing into the else branch line 159. Then we don't have to query the value from hashmap and have it more localized (plus much cheaper execution).

Totally agree with logic of 'direction' and 'ignore_outgoing' implemented by this changeset.